cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -1798,6 +1798,11 @@ Slice::CsGenerator::validateMetadata(const UnitPtr& u)
             const string msg = "'cs:namespace' is deprecated; use 'cs:identifier' to remap modules instead";
             p->unit()->warning(metadata->file(), metadata->line(), Deprecated, msg);
 
+            if (auto cont = dynamic_pointer_cast<Contained>(p); cont && cont->hasMetadata("cs:identifier"))
+            {
+                return "A Slice element can only have one of 'cs:namespace' and 'cs:identifier' applied to it";
```

Comment:
Is it better to say module here as it's the only thing it can be?
```suggestion
                return "A Slice module can only have one of 'cs:namespace' and 'cs:identifier' applied to it";
```

Reply:
The issue is that at this stage of validation, we don't actually know what the type is. It might not be a module.
It could be anything. Maybe the user thinks they need to put `java:package` on everything and incorrectly writes:
```slice
["java:package:com.austininc"]
module HelloThere
{
    ["java:package:com.austininc"]
    ["java:identifier:FailureException"]
    exception Failure {} 
}
```
It would be weird and misleading for the error to talk about modules, when the real problem is with the exception!